build.rs

```diff
@@ -99,6 +101,32 @@ fn fetch_libipt() {
     env::set_current_dir(&Path::new("..")).unwrap();
 }
 
+// Checks if the CPU supports Intel Processor Trace.
+// We use this to decide whether to run the perf_pt backend tests. Although this would be better as
+// a runtime check, it's OK since we won't distribute the test binary.
```

Comment:
I don't think this could ever sensibly be a runtime check (after all, we'll statically compile in the backend!), so we probably delete this comment.

Reply:
That's not right. We could build the our code on a system with a CPU with no PT support and distribute it to a system with a CPU that does support PT. Otherwise I would have re-used the existing `perf_pt` compile time flag.

Reply:
Dumb question: why can't we use the `perf_pt` flag?

Reply:
That decides if we are going to build with the `perf_pt` backend. The prerequisite is that the build system has a new enough Linux Perf API. It says nothing about whether the CPU on the build system supports PT or not.

Reply:
Is it ever useful to say "perf is too old, but we're going to do stuff with PT support anyway"?

Reply:
If perf is too old the PT backend cannot be built and the tests cannot be run.

Reply:
Yes, I understand that, but it doesn't quite answer my question. Basically, will we ever make use of a situation where `perf_pt == false` but this function you've added returns true?

Reply:
Nope. But we do need a separate flag for the case where Perf is new enough, but the CPU on the build machine doesn't have PT. Do you follow?

Reply:
Ah, yes. I wonder if we could just collapse that into one flag though, since we don't *really* care about the individual parts?

Reply:
We could collapse it into one, but it would mean that we can't distribute binaries if the build system has no PT on the chip.

Reply:
Aha, OK, you've convinced me!